feat: add findSubMatch behavior (#250)
updatecli/website: content/docs/core/transformer.adoc

| to | &#10004;|-| "to" value defines the string that we want to have
|===

=== FindSubMatch

> A findSubMatch rule capture the information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).

Reviewer: ```suggestion
A findSubMatch rule captures the desired information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).
```

| to | &#10004;|-| "to" value defines the string that we want to have
|===

=== FindSubMatch

A findSubMatch rule captures the desired information by applying "pattern" as a regexp and using "captureIndex" value as the index for the capture to return (default 0).

[cols="1,1,1,4",options=header]
|===
| Name | Required | Default |Description
| pattern | &#10004;|-| "pattern" value defines the regular expression to apply

> | captureIndex |-|-| "captureIndex" value defines the index of the capture to return (0 by default)

Reviewer: As discussed together, add explanation about "0" returning all submatches, and individual submatch indexes starting at 1

(In the example below, `captureIndex: 1` returns the `(.*?)` submatch)

Author: awesome but I don't have the button to commit the suggestion ...

Reviewer: It's a simple comment, not a suggestion, you'll need to go to the "files" tab and create a new suggestion from it, or copy/paste it in your local repo and commit then push it.

Note: this should also be added as comment in updatecli/updatecli corresponding code

Reviewer: @smerle33 I've added the suggestions and the PR ;)

> | captureIndex |-|-| "captureIndex" value defines the index of the capture to return (0 by default)

Reviewer: ```suggestion
| captureIndex |-|-| "captureIndex" value defines the index of the capture to return. Note that the default value `0` returns all submatches, and individual submatch indexes start at `1`.
```

|===

== Example

> 

Reviewer: ```suggestion
The `findSubmatch` below will return the submatch captured with `(.*?)`:
```

      - replacers:
          - from: "-jdk15"
            to: "-jdk17"
      - findsubmatch:

>           pattern: '"(.*?):'

Reviewer: I would use an example with multiple possible sub matches here.

Reviewer 2: Then 2 examples :)

Author: I updated this one, 2 exemples are not compliant with what's existing

      - replacers:
          - from: "-jdk15"
            to: "-jdk17"
      - findsubmatch:

>           pattern: '"(.*?):(.*)v'

Reviewer: ```suggestion
          pattern: '"(.*?):(.*)'
```
